src/main-view/content/row-settings/send-keys-row-settings.js

```diff
@@ -145,6 +179,60 @@ class SendKeysRowSettings extends Gtk.Grid {
     this.show_all();
   }
 
+  hasShortcut() {
+    return this.keys.length > 0 || this.modifiers.length > 0;
+  }
+
+  clearShortcut() {
+    this.keys = [];
+    this.modifiers = [];
+    this.modifiersEntry.text = '';
+    this.keysEntry.text = ''
+  }
+
+  onKeyPressed(widget, event) {
+    
+    // if no grab, no key/modifier registering
+    if(!widget.has_grab()){
+      return;
+    }
+
+    var keyval = event.get_keyval()[1]
+    var name = Gdk.keyval_name(keyval)
+
+    /// if escape, clear
+    if (keyval == Gdk.KEY_Escape) {
+      this.clearShortcut();
+      return;
+    }
+    // if modifier
+    if (MODIFIERS.includes(name)) {
+      // did not add new if already exists in list
+      if (this.modifiers.includes(name)) {
+        return;
+      }
+      this.modifiers.push(name);
+      this.modifiersEntry.text = this.modifiers + '';
+    } else {
+      if (this.keys.includes(name)) {
+        return;
+      }
+      this.keys.push(name);
+      this.keysEntry.text = this.keys + '';
+    }
+
+    this.buildShortcutLabelContent();
+  }
+
+  buildShortcutLabelContent() {
+    if (!this.hasShortcut()) {
+      this.shortcutButton.label = _('Click here to add shortcut');
+    } else {
+      this.shortcutButton.label = (this.modifiers.concat(this.keys) + '').replaceAll(',', '+');
```

Comment:
I'm getting a crash here:

```
(com.github.joseexposito.touche:14354): Gjs-WARNING **: 18:20:28.139: JS ERROR: TypeError: (this.modifiers.concat(...) + "").replaceAll is not a function
```

Is it equivalent to...
```suggestion
      this.shortcutButton.label = [...this.modifiers, ...this.keys].replace(/,/g, '+');
```